definitions/npm/detox_v7.0.x/test_detox_v7.0.x.js

```diff
@@ -0,0 +1,170 @@
+// $ExpectError
+device.launchApp({permissions: false, url: false});
+
+// $ExpectError
+device.terminateApp({err: 'err'})
+device.terminateApp()
+device.terminateApp('other.bundle.id')
+device.reloadReactNative()
+device.sendToHome()
+device.installApp()
+device.uninstallApp()
+// $ExpectError
+device.openURL({url: 'https://flow.org', sourceApp: false})
+device.openURL({url: 'https://flow.org'})
+
+// $ExpectError
+device.sendUserNotification()
+device.sendUserNotification({})
+
+// $ExpectError
+device.setOrientation('normal')
+device.setOrientation('portrait')
+
+// $ExpectError
+device.setLocation()
+device.setLocation(32.04, 34.78)
+
+device.setURLBlacklist(['https://flow.org','https://flow.org'])
+device.enableSynchronization()
+device.disableSynchronization()
+device.resetContentAndSettings()
+device.getPlatform()
+
+
+// Matchers
+// --------
+by.id('flowType')
+by.text('flowType is great!')
+by.label('welcome')
+// $ExpectError
+by.label(123)
+by.type('RCTImageView')
+by.traits(['button'])
+// $ExpectError
+by.traits('button')
+
+
+// Actions on element
+// ------------------
+ element(by.id('testElement')).tap()
+ element(by.id('testElement')).longPress()
+
+// $ExpectError
+ element(by.id('testElement')).multiTap(true)
+ element(by.id('testElement')).multiTap(3)
+
+// $ExpectError
+ element(by.id('testElement')).tapAtPoint()
+ element(by.id('testElement')).tapAtPoint({x: 5, y: 10})
+
+// $ExpectError
+ element(by.id('testElement')).typeText(false)
+ element(by.id('testElement')).typeText('type this')
+
+// $ExpectError
+ element(by.id('testElement')).replaceText()
+ element(by.id('testElement')).replaceText('text')
+
+ element(by.id('testElement')).clearText()
+
+// $ExpectError
+ element(by.id('testElement')).scroll()
+ element(by.id('testElement')).scroll(123, 'down')
+
+// $ExpectError
+ element(by.id('testElement')).scrollTo('down')
+ element(by.id('testElement')).scrollTo('left')
+
+// $ExpectError
+ element(by.id('testElement')).swipe('sideways', 'medium', 0.5)
+ element(by.id('testElement')).swipe('down', 'fast', 0.5)
+
+
+// Expect API
+// ----------
+expect(element(by.id('UniqueId204'))).toBeVisible()
+expect(element(by.id('UniqueId204'))).toBeNotVisible()
+expect(element(by.id('UniqueId204'))).toBeVisible()
+expect(element(by.id('UniqueId204'))).toExist()
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveId(123)
+expect(element(by.id('UniqueId204'))).toHaveId('someTestID')
+
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveLabel()
+expect(element(by.id('UniqueId204'))).toHaveLabel('testLabel')
+
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveText(123)
+expect(element(by.id('UniqueId204'))).toHaveText('some text')
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveValue(true)
+expect(element(by.id('UniqueId204'))).toHaveValue("1")
+expect(element(by.id('UniqueId204'))).toNotExist()
+
+// waitFor API
+waitFor(element(by.id('UniqueId204'))).toBeNotVisible()
```

Comment:
waitFor only works with `withTimeout(Number)`, the below lines will never trigger anything. is there a way to enforce it with flow ?

Reply:
I cannot really see anything we can do here using Flow.